src/correlationUtils.ts

```diff
@@ -0,0 +1,80 @@
+import { DateTime } from "luxon";
+import { DataType } from "./interfaces";
+
+export const splitAndTrim = (fields: string): string[] => {
+	if (fields === "") return [];
+	else return fields.split(",").map((str) => str.trim());
+};
+
+export const getInnerValue = (value: any) => {
+    const unproxied = unproxy(value);
+    if (unproxied.length === 1) {
+        if (typeof unproxied[0] === "string") {
+            let list = unproxied[0];
+            if (list.startsWith("[") && list.endsWith("]")) {
+                list = list.slice(1, -1);
+            }
+            const splits = splitAndTrim(list).map((item) => {
+                if (item.startsWith(`"`) && item.endsWith(`"`)) {
+                    return item.slice(1, -1);
+                } else return item;
+            });
+            if (splits.length === 1) return splits[0];
+            else return splits;
+        } else {
+            if (unproxied[0].type === "file") {
+                return unproxied[0].path;
+            } else return unproxied[0];
+        }
+    } else {
+        if (unproxied[0].type === "file") {
+            return unproxied.map((link) => link.path);
+        } else return unproxied;
+    }
+}
+
+export const unproxy = (item: any): DataType[] => {
+    const unproxied = [];
+
+    const queue = [item];
+    while (queue.length) {
+        const currItem = queue.shift();
+        // "Proxy" for checking if `currItem` is a proxy
+        if (typeof currItem.defaultComparator === "function") {
+            const possibleUnproxied = Object.assign({}, currItem);
+            const { values } = possibleUnproxied;
+            if (values) queue.push(...values);
+            else unproxied.push(possibleUnproxied);
+        } else unproxied.push(currItem);
+    }
+    return unproxied;
+}
+
+export const processPages = (pages: { [field: string]: any }[],fieldsToCheck: string[], dates: DateTime[]): {pages: { [field: string]: any }[], dates: DateTime[]} => {
+    pages.forEach((page) => {
+        const potentialDate = DateTime.fromISO(page.file.name);
+        if (potentialDate.isValid) dates.push(potentialDate);
+
+        fieldsToCheck.forEach((field) => {
+            const value = page[field];
+            if (value) {
+                page["hasFieldsOfInterest"] = true;
+                page[field] = getInnerValue(value);
+            }
+        });
+    });
+
+    return {pages: pages.filter((page) => page.hasFieldsOfInterest), dates: dates}
+}
```

Comment:
Why not initialise `dates` inside this function?
Is there a case where we would pass in a non-empty list of dates?

Reply:
I agree, I'll make the change.